### DISCUSSION TITLE
ENTER and SHIFT+ENTER behaviour

### DISCUSSION THREAD

**Discussion by himadridev** [mdx-editor/editor#352] [Q&A]
Thank you for building this awesome library 👏🏼 

I have a use-case of a chat interface, where user input field is markdown editor (similar to slack input field). As per requirement, user should be able to submit the message on ENTER press and to break new line user has to use SHIFT+ENTER.

I have created a plugin and registered KEY_ENTER_COMMAND to capture & handle the ENTER 👇🏼 
```
  useEffect(() => {
    return editor.registerCommand<KeyboardEvent>(
      KEY_ENTER_COMMAND,
      (event: KeyboardEvent) => {
        if (event.key === 'Enter' && !event.shiftKey) {
          event.preventDefault();
  
          onSendMessage?.();
          return true;
        }

        return false;
      },
      COMMAND_PRIORITY_LOW,
    );
  }, []);
```

But with this approach, editor's line break for new formatting stopped _(using SHIFT+ENTER for new line)_ 👇🏼 

https://github.com/mdx-editor/editor/assets/23113952/0ba0a0db-14eb-4428-8d08-6bbb931fc136

That's because my custom command handler blocking the expected ENTER press for the editor.

Q: **Is there any way to use SHIFT+ENTER for line break for new formatting?** So that I can use ENTER for submitting the message.

**Comment by himadridev** [comment]
@petyosi Thanks for starting a discussion. Any thoughts on this?

**Reply by petyosi** [reply]
No, no thoughts. This is entirely in the space of the Lexical runtime and not constrained by the MDXEditor API.

**Comment by joesaunderson** [comment]
@himadridev did you come up with a solution for this? Hitting the same thing

**Comment by eaolukoya** [comment]
Try registering INSERT_PARAGRAPH_COMMAND instead of KEY_ENTER_COMMAND in your plugin
